flocker/node/agents/testtools.py

```diff
@@ -104,11 +104,46 @@ def cinder_client_from_environment():
             'for details of the expected format.'
         )
 
-    config = yaml.load(config_file.read())
-    provider_name = os.environ.get('CLOUD_PROVIDER', DEFAULT_CLOUD_PROVIDER)
-    provider_config = config[provider_name]
-    cinder_client_factory = CINDER_CLIENT_FACTORIES[provider_name]
-    return cinder_client_factory(**provider_config)
+    config = yaml.safe_load(config_file.read())
+    return config
+
+
+def ec2_client_from_environment():
+    """
+    Create an EC2 Boto client using credentials from a config
+    file path which may be supplied as an environment variable.
+    See ``load_config`` for details on where config is populated from.
+
+    :returns: An instance of EC2 Boto client
+        using provider specific credentials found in ``CLOUD_CONFIG_FILE``.
+    """
+    config = load_client_config_from_environment()
+    provider_name = os.environ.get('CLOUD_PROVIDER')
```

Comment:
Perhaps this bit could be moved into `load_client_config_from_environment` and supply that function with a provider name.